sgkit/io/vcf/vcf_reader.py

```diff
@@ -49,6 +64,163 @@ def get_region_start(region: str) -> int:
+        vcf_number = field_def.get("Number", vcf_field_defs[key]["Number"])
+        dimension, size = vcf_number_to_dimension_and_size(
+            vcf_number, category, key, field_def, DEFAULT_MAX_ALT_ALLELES
+        )
+        vcf_type = field_def.get("Type", vcf_field_defs[key]["Type"])
+        description = field_def.get(
+            "Description", vcf_field_defs[key]["Description"].strip('"')
+        )
+        dtype, fill_value = _vcf_type_to_numpy_type_and_fill_value(
+            vcf_type, category, key
+        )
+        chunksize: Tuple[int, ...]
+        if category == "INFO":
+            variable_name = f"variant_{key}"
+            dims = [DIM_VARIANT]
+            chunksize = (chunk_length,)
+        elif category == "FORMAT":
+            variable_name = f"call_{key}"
+            dims = [DIM_VARIANT, DIM_SAMPLE]
+            n_sample = len(vcf.samples)
+            chunksize = (chunk_length, n_sample)
+        if dimension is not None:
+            dims.append(dimension)
+            chunksize += (size,)
+
+        array = np.full(chunksize, fill_value, dtype=dtype)
+
+        return cls(
+            category,
+            key,
+            variable_name,
+            description,
+            dims,
+            fill_value,
+            array,
+        )
+
+    def add_variant(self, i: int, variant: Any) -> None:
+        if self.category == "INFO":
+            val = variant.INFO.get(self.key)
+            if val is not None:
+                assert self.array.ndim in (1, 2)
+                if self.array.ndim == 1:
+                    self.array[i] = val
+                elif self.array.ndim == 2:
+                    self.array[i] = self.fill_value
+                    a = np.array(val, dtype=self.array.dtype)
+                    if a.ndim == 0:
+                        a = a.reshape(1)  # ensure 1D
+                    a = a[: self.array.shape[-1]]  # trim to fit
+                    self.array[i, : a.shape[-1]] = a
+            else:
+                self.array[i] = self.fill_value
+        elif self.category == "FORMAT":
+            val = variant.format(self.key)
+            if val is not None:
+                assert self.array.ndim in (2, 3)
+                if self.array.ndim == 2:
+                    self.array[i] = val[..., 0]
+                elif self.array.ndim == 3:
+                    self.array[i] = self.fill_value
+                    a = val
+                    a = a[..., : self.array.shape[-1]]  # trim to fit
+                    self.array[i, ..., : a.shape[-1]] = a
+            else:
+                self.array[i] = self.fill_value
+
+    def truncate_array(self, length: int) -> None:
+        self.array = self.array[:length]
+
+    def update_dataset(
+        self, ds: xr.Dataset, add_str_max_length_attrs: bool = False
+    ) -> None:
+        # cyvcf2 represents missing Integer values as the minimum int32 value
```

Comment:
I'm slightly worried about this - could we be silently altering data? Suppose someone uses INT32_MIN in their data, won't we silently change this to fill_value then? I guess this is an upstream problem with cyvcf2.

Reply:
I opened https://github.com/brentp/cyvcf2/issues/195 for this.

Reply:
I don't think they could use INT32_MIN for anything else as this value is explicitly interpreted as a 'missing' integer in the [VCF spec](https://samtools.github.io/hts-specs/VCFv4.3.pdf) (page 30)